### PR TITLE
'getting started' vignette

### LINKED ISSUE #17: Create vignette for tar_terra_* targets use
See https://github.com/ropensci-books/targets/blob/main/walkthrough.qmd and https://github.com/ropensci/stantargets/blob/main/vignettes/introduction.Rmd for details on setting up targets in a vignette

### PR DESCRIPTION
**Pull request by njtierney**
Still need to fill this in with more content from issues.

For example

- using gdam and vsizip and cgaz https://github.com/njtierney/geotargets/issues/22
- using `geotargets_option_set` - https://github.com/njtierney/geotargets/pull/45

This will resolve #17

### CODE REVIEW

File: vignettes/using-terra.Rmd [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      terra::project(
+        terra::rast(system.file("ex", "elev.tif", package = "terra")) * z,
+        projection
+      )
+    }
+    list(
+      tar_terra_sprc(
+        raster_elevs,
+        # two rasters, one unaltered, one scaled by factor of 2 and
+        # reprojected to interrupted good homolosine
```
Comment (Aariq) [inline vignettes/using-terra.Rmd:115]:
```suggestion
        # reprojected to interrupted Goode homolosine
```

File: vignettes/using-terra.Rmd [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    }
+    list(
+      tar_terra_vect(
+        terra_vect_example,
+        lux_area()
+      )
+    )
+  })
+  tar_make()
+  x <- tar_read(terra_rast_example)
```
Comment (Aariq) [inline vignettes/using-terra.Rmd:91]:
```suggestion
  x <- tar_read(terra_vect_example)
```

File: vignettes/using-terra.Rmd [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        command = terra::sprc(list(
+          elev_scale(1),
+          elev_scale(2, "+proj=igh")
+        ))
+      )
+    )
+  })
+  targets::tar_make()
+  x <- targets::tar_read(raster_elevs)
+})
```
Comment (Aariq) [inline vignettes/using-terra.Rmd:125]:
```suggestion
  x <- targets::tar_read(raster_elevs)
  x
})
```

**Review by Aariq — COMMENTED** [review]
It's a good start!  I think eventually we should add an example of how to change file type and pass GDAL options, but that could be in a separate PR if you'd rather.

**Comment by Aariq** [comment]
Not sure why, but when I `build_vignette()` and then view it with `vignette("using-terra")`, I don't see any output, just code.

**Comment by njtierney** [comment]
Good idea with precomputing!

**Comment by Aariq** [comment]
Because of the trick to set `root.dir` to a temporary directory, you don't need `tar_dir()` at all I think.  All examples have three chunks: one with echo: false to write the _targets.R file with `tar_script()`, one with eval: false to show the contents of _targets.R, and one with echo and eval true to run the pipeline and inspect results.

I didn't include much about the raster and vector GDAL options, because I don't have a good motivating example, but I did include examples of changing the driver and added an example with `preserve_metadata =  "zip"`.

**Comment by Aariq** [comment]
@njtierney, I think this is pretty much done.  Can you take a look when you get a chance?

**Comment by njtierney** [comment]
Thanks so much for this, @Aariq !

I think it's great. I think that down the track we could provide some more information about using things like `pattern = map()`, but that might be best put in another future vignette, or perhaps link to a repository that contains a worked example.

Happy to merge!

**Comment by Aariq** [comment]
Yeah, the plan was to show some dynamic branching examples in a separate vignette focused on `tar_terra_tile()` (#111), but I'm stuck on that because I really wanted `tile_blocksize()` to work (#99)!

**Pull request merged**